grl/environment/tmaze_lib.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+
+def tmaze(n: int, discount: float = 0.9):
+    """
+    Return T, R, gamma, p0 and phi for tmaze, for a given corridor length n
+
+                            +---+
+          X                 | G |     S: Start
+        +---+---+---+   +---+---+     X: Goal Indicator
+        | S |   |   |...|   | J |     J: Junction
+        +---+---+---+   +---+---+     G: Terminal (Goal)
+          0   1   2       n | T |     T: Terminal (Non-Goal)
+                            +---+
+    """
+    n_states = 2 * n # Corridor
+    n_states += 2 # Start
+    n_states += 2 # Junction
+    n_states += 1 # Terminal state (includes both goal/non-goal)
+
+    T_up = np.eye(n_states, n_states)
+    T_down = T_up.copy()
+    T_up[[-1, -2, -3], [-1, -2, -3]] = 0
+    T_down[[-1, -2, -3], [-1, -2, -3]] = 0
+
+    # If we go up or down at the junctions, we terminate
+    T_up[[-1 - 1, -2 - 1], [-1, -1]] = 1
+    T_down[[-1 - 1, -2 - 1], [-1, -1]] = 1
```

Comment:
Simplify math expressions: `[-2, -3]`. Even better would be to set constants like `JUNCTION_GOAL_UP` and `JUNCTION_GOAL_DOWN`